text/0000-file-open-with.md

```diff
@@ -0,0 +1,116 @@
+- Feature Name: file_open_with
+- Start Date: 2018-11-08
+- RFC PR: 
+- Rust Issue: #55762
+
+# Summary
+[summary]: #summary
+
+This RFC proposes making `File::open()` consistent by deprecating `OpenOptions::new().read(true).write(true).open("existing_file")` and adding `File::open_with("existing_file", OpenOptions::new().read().write())` instead.
+
+# Motivation
+[motivation]: #motivation
+
+The current way to open an existing file in read-only mode is this:
+
+    File::open("foo.txt")
+
+And to create a new file it is:
+
+    File::create("foo.txt")
+
+But if you want to open an existing file in read/write mode it is this:
+
+use std::fs::OpenOptions;
+
+    let mut file = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .open("foo.txt");
+
+This is inconsistent and unexpected. I propose that we deprecate `OpenOptions::open()` and add `File::open_with(path: &str, options: &OpenOptions)`.
+
+    let mut file = File::open_with("foo.txt", OpenOptions::new().read().write());
+
+This matches the normal way of doing "call a method with some options" in Rust, for example `TcpStream::connect(addr)` and `TcpStream::connect_timeout(addr, timeout)`.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Reading and Writing Files
+
+To open a file in read-only mode you can use `File::open()` like this:
+
+    let mut file = File::open("existing_file");
+
+This will open the file in read-only mode. To create a new file and write to it you can use:
+
+    let mut file = File::create("new_file");
+
+This will open the file in read/write mode and create it if it doesn't already exist. If you want to open an existing file, but *not* create it if it already exists, use this code:
+
+    let mut file = File::open_with("existing_file", OpenOptions::new().read().write());
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Internally this could be implemented exactly like this:
+
+```
+impl File {
+  pub fn open_with(filename: &str, options: &OpenOptions) -> File {
```

Comment:
This will need to take the same `AsRef<Path>` type parameter as File::open.<br>https://doc.rust-lang.org/std/fs/struct.File.html#method.open